plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiConfig.java

```diff
@@ -0,0 +1,240 @@
+    private int partitionScannerParallelism = 16;
+    private int splitGeneratorParallelism = 16;
+    private int minPartitionBatchSize = 10;
+    private int maxPartitionBatchSize = 100;
+    private boolean sizeBasedSplitWeightsEnabled = true;
+    private DataSize standardSplitWeightSize = DataSize.of(128, MEGABYTE);
+    private double minimumAssignedSplitWeight = 0.05;
+    private int maxSplitsPerSecond = Integer.MAX_VALUE;
+    private int maxOutstandingSplits = 1000;
+
+    public List<String> getColumnsToHide()
+    {
+        return columnsToHide;
+    }
+
+    @Config("hudi.columns-to-hide")
+    @ConfigDescription("List of column names that will be hidden from the query output. " +
+            "It can be used to hide Hudi meta fields. By default, no fields are hidden.")
+    public HudiConfig setColumnsToHide(String columnsToHide)
+    {
+        this.columnsToHide = COMMA_SPLITTER.splitToStream(nullToEmpty(columnsToHide))
+                .map(s -> s.toLowerCase(ENGLISH))
+                .collect(toImmutableList());
+        return this;
+    }
+
+    @Config("hudi.metadata-enabled")
+    @ConfigDescription("Fetch the list of file names and sizes from metadata rather than storage.")
+    public HudiConfig setMetadataEnabled(boolean metadataEnabled)
+    {
+        this.metadataEnabled = metadataEnabled;
+        return this;
+    }
+
+    public boolean isMetadataEnabled()
+    {
+        return this.metadataEnabled;
+    }
+
+    @Config("hudi.use-metastore-for-partitions")
+    @ConfigDescription("By default, partition info is fetched from the metastore. " +
+            "When this config is disabled, then the partition info is fetched using Hudi's partition extractor and relative partition path.")
+    public HudiConfig setUseMetastoreForPartitions(boolean useMetastoreForPartitions)
+    {
+        this.useMetastoreForPartitions = useMetastoreForPartitions;
+        return this;
+    }
+
+    public boolean getUseMetastoreForPartitions()
+    {
+        return this.useMetastoreForPartitions;
+    }
+
+    @Config("hudi.parquet.use-column-names")
+    @ConfigDescription("Access Parquet columns using names from the file. If disabled, then columns are accessed using index."
+            + "Only applicable to Parquet file format.")
+    public HudiConfig setUseParquetColumnNames(boolean shouldUseParquetColumnNames)
+    {
+        this.shouldUseParquetColumnNames = shouldUseParquetColumnNames;
+        return this;
+    }
+
+    public boolean getUseParquetColumnNames()
+    {
+        return this.shouldUseParquetColumnNames;
+    }
+
+    @Config("hudi.partition-scanner-parallelism")
+    @ConfigDescription("Number of threads to use for partition scanners.")
+    public HudiConfig setPartitionScannerParallelism(int partitionScannerParallelism)
+    {
+        this.partitionScannerParallelism = partitionScannerParallelism;
+        return this;
+    }
+
+    @Min(1)
+    @Max(100)
+    public int getPartitionScannerParallelism()
```

Comment:
I don't seem to see this setting used anywhere in the code.
Can we remove it and add it only when it will be actually having a purpose?